backend/src/apiserver/server/experiment_server.go

```diff
@@ -29,6 +37,11 @@ func (s *ExperimentServer) CreateExperiment(ctx context.Context, request *api.Cr
 
 func (s *ExperimentServer) GetExperiment(ctx context.Context, request *api.GetExperimentRequest) (
 	*api.Experiment, error) {
+	err := s.canAccessExperiment(ctx, request.Id)
```

Comment:
[perf] Will this method send a request to get experiment? We will get experiment again in the following command.
So does this method always get experiment twice?

I'm not sure if there are any caching layers to avoid that, just want to confirm what's current situation and shall we improve on this?

Reply:
Added to TODO.

Reply:
The plan SGTM